src/main/java/spoon/reflect/visitor/LiteralHelper.java

```diff
@@ -58,12 +58,11 @@ private static String getBasedString(Double value, LiteralBase base) {
 
 	/**
 	 * @param literal CtTextBlock to be converted
-	 * @param numTabs
 	 * @return source code representation of the literal
 	 */
 	public static String getTextBlockToken(CtTextBlock literal) {
 		String token = "\"\"\"\n"
-				+ literal.getValue()
+				+ literal.getValue().replaceAll("\\\\", "\\\\\\\\")
```

Comment:
Why does this work?

Reply:
It escapes each single backslash by replacing it with a double backslash. It just looks weird because each backslash has to be escaped twice, once for the string literal and then again in the regular expression.

Reply:
> It escapes each single backslash by replacing it with a double backslash.

Would it not be sufficient to write `.replaceAll("\". "\\")`? This is what I can infer from your explanation.

Reply:
@algomaster99 no, that's a compile time error. The backslash inside the quotes has to be escaped. Then you have a regex consisting of only a backslash. That is an error because the backslash has to be escaped in the regex. So you need another backslash escaping that one, which has to be escaped inside the quotes. That's why you need four backslashes to represent a single backslash in the search regex.

The same then applies for the replacement, so this makes it four backslashes in the search string and eight backslashes in the replacement string.

But you have a point in that we don't need really a regex here. So we can use `String.replace(CharSequence, CharSequence)` instead (note the different method name), which does not use regular expressions. This will be both faster and require only half the escaping. I have updated the PR.